### DISCUSSION TITLE
Data security with topic representation APIs

### DISCUSSION THREAD

**Discussion by AshokNagula** [MaartenGr/BERTopic#1052] [Q&A]
As  topic representations introduced with the latest version. how secure  the  documents (data) are when we use third party APIs like Cohere or OpenAI. Is bertopic sending the documents to API providers servers or its leveraging the models of these providers .

**Comment by MaartenGr** [comment]
To a certain extent, that has been taken into account when developing the API models. In those models, there is the option to customize your prompts with the `"[DOCUMENTS]"` and `"[KEYWORDS]"` that allow for controlling which data is sent to those APIs. 

It works as follows, let's say you have the following, customized prompt:

```python
prompt = """
I have topic that contains the following documents: [DOCUMENTS]. 
The topic is described by the following keywords: [KEYWORDS].
Based on the above information, can you give a short label of the topic?
"""
```

In this prompt, the `"[DOCUMENTS]"` tag will be replaced with 3 documents that are most representative of a topic. If you have sensitive data, it might be best **not** using this tag if you want to prevent sharing your data. In contrast, the `"[KEYWORDS]"` tag are simply the top 10 keywords for a topic based on the c-TF-IDF representation. Since these keywords are in a sense aggregated information, they typically do not convey sensitive information. 

In other words, it depends on the prompt and tags which information is sent to the server. 

**NOTE**: The default prompt can be retrieved with  `representation_model.default_prompt_`.
